lib/fetch/index.js

```diff
@@ -1344,7 +1344,9 @@ async function httpNetworkOrCacheFetch (
   //    user agents should append `User-Agent`/default `User-Agent` value to
   //    httpRequest’s header list.
   if (!httpRequest.headersList.contains('user-agent')) {
-    httpRequest.headersList.append('user-agent', __filename.endsWith('index.js') ? 'undici' : 'node')
+    // __commonJS is injected by esbuild when bundled. The esbuild bundle is used in node core.
+    // Therefore if we're running the esbuild bundle, we're likely in node core.
+    httpRequest.headersList.append('user-agent', typeof __commonJS !== 'function' ? 'undici' : 'node')
```

Comment:
not a fan of relying on an unspecified global. Someone could decide to set this themselves and undici would break itself.

Reply:
fwiw it's also possible to set __filename

Reply:
true but there is at least an assumption/definition for `__filename`. If someone decides to mess with that - it's on them.